src/extension.ts

```diff
@@ -520,7 +530,12 @@ located next to Run / Debug Tests toolbar buttons.`);
       await model.workspaceChanged(change);
     // Workspace change can be deferred, make sure editors are
     // decorated.
-    this._updateVisibleEditorItems();
+    await this._updateVisibleEditorItems();
+    this._watchSupport.workspaceChanged(change);
+  }
+
+  private _watchesTriggered(watches: Watch[]) {
+    this._runMatchingTests(watches, 'watch');
```

Comment:
We probably should enqueue watch runs.

Reply:
I'll queue them in a promise chain for now.